### DISCUSSION TITLE
Quality of life features for the skin layout editor

### DISCUSSION THREAD

**Discussion by FriendsNone** [ppy/osu#22347] [Ideas] [answered]
With the recent skin components additions in 2023.121.0, I think the editor needs some QoL features to make customizing it a bit more sane.

Such as:
- Keyboard shortcuts
- Undo and Redo changes
- Copy and Paste components
- Grid and/or Guide lines
- Snap to grid/guide/component
- Uniform transform using height/width handles
- Inputs (like a textbox) for position and size
- Revert to last saved or Discard changes option

Those are just some QoL features that I could think of, they're somewhat inspired from Photoshop.

**Comment by Theighlin** [comment, marked as answer]
The skin editor is a work in progress and some of these features have already been suggested. 
This discussion has too large of a scope to be actionable.

You're welcome to split each of these to their own discussion, after having searched existing ones as to not create duplicates.
